### PR TITLE
[New] Add ng-browser-info w/ git auto-update via single package.json

### LINKED ISSUE #11890: [Request] Add ng-browser-info
**Library name:** ng-browser-info
**Repository url:** https://github.com/transferwise/ng-browser-info
**npm package url:** none
**License:** [Apache-2.0](https://github.com/transferwise/ng-browser-info/blob/master/LICENSE)
**Official homepage:** http://transferwise.github.io/ng-browser-info/

## Something you might want to know about CDNJS
CDNJS provides a CDN containing front-end libraries to allow all the web developers from around the world to use the libraries on our free CDN without any additional download/upload process.

## How to deal with this issue?
You can choose to add the library via single package.json or with its required assets. As for the working environment, you can choose to work on the GitHub GUI or on your computer (locally).
Working locally will let you learn git (version control) and debug skills. However, it needs more spaces on your computer in comparison with using GitHub. So don't feel any pressure if you want to use the GitHub GUI. If you decide to work locally, it's recommend to pulling cdnjs repository with [sparseCheckout](https://github.com/cdnjs/cdnjs/blob/master/documents/sparseCheckout.md ) because the cdnjs repo is too huge.

The followings are the useful steps to let you add the requested library with the GitHub GUI  in an easier way. :muscle:
### Via single package.json
1. [Fork](https://github.com/cdnjs/cdnjs/fork) the cdnjs repo and do the following steps on your forked repo.
2. Create a new, non-master branch by clicking the "Branch: master" dropdown and enter the name of the library being added.
3. Create the file `<lib_name>/package.json` under `ajax/libs` directory by clicking "Create new file" button. You will see the part of the screen like this image 
Find the package.json (or bower.json if there is no package.json) in source repo (including npm). Please copy its content and modify the content if the content doesn't match the format of package.json in cdnjs. There are 9 fields we may need in total here. Other than these fields, please remove them.
* Necessary ones: `name`, `filename`, `description`, `keywords`, `repository`, `author`, `license` and "npm auto-update" (or "git auto-update")
  * `name`: library name, please use the GitHub repo name if the source is GitHub, please use npm package name if the source is npm.
  * `filename`: Find the main file of this library, you might be able to find it after reading its documentation. If the main file is not minified, please still use `<filename>.min.js` or `<filename>.min.css` structure naming, we'll do the minify job.
  * `author`: There might be more than one author. Please use `authors` instead when there are multiple authors.
  * `license`: There also might be more than one license. Please use `licenses` when there are multiple licenses.
  * npm auto-update or git auto-update: Please follow https://github.com/cdnjs/cdnjs/blob/master/documents/autoupdate.md. There are a few rules to help you choose GitHub or npm to be the source. The simplest rule you can use here is to choose the one with more versions. If the number of versions are the same, please use npm because we prefer it.
* Unnecessary: `homepage` if it's the same as the source repo or just a demo page with no link to its source repo.

  Last, please use [JSONLint](http://jsonlint.com/) to validate your package.json. 
4. Commit new file: 
* The heading of the commit message
  * If the source is git: `Add <lib_name> w/ git auto-update via single package.json`
  * If the source is npm: `Add <lib_name> w/ npm auto-update via single package.json`
* The body of the commit message: `close #<issue_number>, cc @<lib_author>`
5. Submit the pull request in cdnjs repo and fill out the asked information to make sure that everything you can do is good.
[Example 1](https://github.com/extend1994/cdnjs/blob/99b1a5ca8de29b6fa96468d417befa093841bba7/ajax/libs/videojs-ima/package.json), [Example 2](https://github.com/extend1994/cdnjs/blob/91d227960987d08efadc95f7719cb2ab5b9614f7/ajax/libs/gotem/package.json), [Official document](https://github.com/cdnjs/cdnjs/blob/master/CONTRIBUTING.md#e-adding-a-new-library-by-a-single-packagejson)

### With library assets
1. The same as the step 1 in the method "via single package.json".
2. The same as the step 2 in the method "via single package.json".
3. Similar to the the method "via single package.json". But here we need the other necessary field: `version` which is the latest stable version. And please create package.json locally and upload it with step 4. Also, don't forget to use [JSONLint](http://jsonlint.com/) to validate your package.json.
4. Every library should be stored under the `ajax/libs` directory. It has its own sub-directory of `ajax/libs` and each version of the library has its own sub-directory of the library directory name, for example: `/ajax/libs/jquery/3.2.1/`. Find the files from the latest stable version this library needs to run on a browser well (which are also the files you list in `files` field of auto-update config in package.json) and upload them according to their versions. Note every file should be the completely the same as every file in the upstream. Make everything is alright with the command `npm test` if you work locally.
5. Commit the files:
* The heading of the commit message: 
  * If the source is git: `Add <lib_name> w/ git auto-update`
  * If the source is npm: `Add <lib_name> w/ npm auto-update`
* The body of the commit message: `close #<issue_number>, cc @<lib_author>`
6. The same as the step 5 in the method "via single package.json".
[Example](https://github.com/cdnjs/cdnjs/pull/10452/files), [Official document](https://github.com/cdnjs/cdnjs/blob/master/CONTRIBUTING.md#d-adding-a-new-library-with-its-assets)

### Work locally
Some useful commands:
1. Pull the repo from cdnjs repo
```
git pull origin master
```

2. Create a non-master branch from master branch.
```
git checkout -b <lib_name>
```

3. Commit the files, with `-v` you can see more detailed information
```
git commit [-v]
```

4. Revise your last commit message
```
git commit --amend [-v]
```

5. Push to GitHub, add `-f` if you want to update your remote branch.
```
git push <remote_name_you_name_for_your_forked_cdnjs_repo> <branch_name> [-f]
```

Feel free to ask some questions if you have any, we will help you as soon as possible. Now don't hesitate to become a contributor of CDNJS :smile:

## References
For more details you can try to refer [our contributing document](https://github.com/cdnjs/cdnjs/blob/master/CONTRIBUTING.md) or [the contributing document from GitHub](https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project)

### PR DESCRIPTION
**Pull request by bpadair32**
close #11890, cc @transferwise

Pull request for issue: 11890
Related issue(s):

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/transferwise/ng-browser-info
 * Official website (optional, not the repository):
 * NPM package url (optional):
 * License and its reference: Apache-2.0
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 123
   - Count of stars: 33
   - Count of forks: 3
 * NPM download stats (optional):
   - Downloads in the last day:
   - Downloads in the last week:
   - Downloads in the last month:

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [ x] This lib was not found on cdnjs repo
 * [x ] No already exist / duplicated issue and PR
 * [ ] The lib has notable popularity
   * [ ] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 800 downloads stats per month on npm registry
 * [x ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x ] Has valid tags for each versions (for git auto-update)
 * [ x] Auto-update setup
 * [x ] Auto-update target/source is valid.
 * [x ] Auto-update filemap is correct.

# Git commit checklist
 * [ x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [ ] The parent of the commit(s) in the PR is not old than 3 days.
 * [ ] Pull request is sending from a non-master branch with meaningful name.
 * [ ] Separate unrelated changes into different commits.
 * [ ] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [ ] Close corresponding issue in commit message
 * [ ] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/ng-browser-info/package.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  "author": {
+    "name": "Uku Pattak",
+    "email": "uku.pattak@transferwise.com"
+  },
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/transferwise/ng-browser-info"
+  },
+  "keywords": [
+    "client", "info", "useragent", "navigator", "angularjs", "angular", "os", "browser", "screen", "cookies"
```
Comment (sufuf3) [inline ajax/libs/ng-browser-info/package.json]: Let's make the one keyword in one line.
Such as https://github.com/cdnjs/cdnjs/blob/master/ajax/libs/JQuery-Snowfall/package.json#L10-L14
Becuase it is useful to us to read.
Thank you.

**Comment by sufuf3** [comment]
@bpadair32 thanks for contribution.
Would you like to work on this and help improve this PR?
If you have any question, please let us know.

Many thanks.

**Comment by sufuf3** [comment]
@bpadair32 thanks for contribution.
Would you mind working on this and helping improve this PR?
If you have any question, please let us know.

Many thanks.

**Comment by sashberd** [comment]
@sufuf3 I helped with this PR. FYR

File: ajax/libs/ng-browser-info/package.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
+{
+  "name": "ng-browser-info",
+  "filename": "ngBrowserInfo.min.js",
+  "description": "AngularJS service that gives you a collection of methods for knowing more about your client browser ",
```
Comment (sufuf3) [inline ajax/libs/ng-browser-info/package.json]: The description need to as the same with upstream.
From `"AngularJS service that gives you a collection of methods for knowing more about your client browser "` to [`"ngBrowserInfo is an AngularJS service that gives you a collection of methods for knowing more about your client browser."`](https://github.com/transferwise/ng-browser-info/blob/master/package.json#L3)
Thank you.

**Comment by sashberd** [comment]
@sufuf3 I do not see the reason to wait another week for the author`s PR response. He had a half year to do this and it is not happened.
Moreover, why we should close this PR and wait for another contributor to open this issue again(again review->time). Here we have PR that is almost done and I ca n help to fix and merge it. This will reduce number of PR`s and number of issues in cdnjs stack
The PR was updated. FYR

**Comment by sashberd** [comment]
@sufuf3 The commit message was updated. FYR

**Comment by susenjitdey** [comment]
simple question, am trying to use "npm install ng-browser-info" and getting the belwo error, sorry if this is not the right place to question, but I need a solution.
This is the error

npm WARN notice Due to a recent security incident, all user tokens have been invalidated. Please see https://status.npmjs.org/incidents/dn7c1fgrr7ng for more details. To generate a new token, visit https://www.npmjs.com/settings/~/tokens or run "npm login".
npm ERR! code E404
npm ERR! 404 Not Found: ng-browser-info@latest

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\sdey1\AppData\Roaming\npm-cache\_logs\2018-07-25T10_17_01_291Z-debug.log

**Comment by PeterDaveHello** [comment]
@susenjitdey this means the package doesn't exist on npm registry.

**Comment by susenjitdey** [comment]
@PeterDaveHello Thanks, for the info am very new to all this, can you please guide what I am suppose to do.Please guide.

**Comment by susenjitdey** [comment]
@PeterDaveHello is this something that I am expected to perform. npm install ng-browser-info --registry http://registry.npmjs.org/

**Comment by PeterDaveHello** [comment]
@susenjitdey this is not a npm support channel. If you wanna use this library, try: https://cdnjs.com/libraries/ng-browser-info

**Pull request merged**